include/nlohmann/detail/input/parser.hpp

```diff
@@ -479,6 +479,10 @@ class parser
             error_msg += concat(m_lexer.get_error_message(), "; last read: '",
                                 m_lexer.get_token_string(), '\'');
         }
+        else if (last_token == token_type::end_of_input && m_lexer.get_position().chars_read_total == 1)
+        {
+            error_msg += concat("unexpected end of input, check that your input string or stream contains the expected JSON");
```

Comment:
That error message confuses me.

Reply:
Hi again, I'm free for a few more days, my semester starts after that, it'd be great if you could reply in the next 2-3 days, however, I'm still happy to contribute after that.

Reply:
I'm on vacation, so I cannot really help here.

Reply:
I would recommend something along the lines of "the provided input was empty, check that you are providing the expected input"